server.go

```diff
@@ -148,38 +208,69 @@ func (srv *Server) Publish(channels []string, ev Event) {
 	}
 }
 
-// PublishComment publishes a comment to one or more channels.
-func (srv *Server) PublishComment(channels []string, text string) {
+// PublishWithAcknowledgment publishes an event to one or more channels, returning a channel that will receive
+// a value after the event has been processed by the server.
+//
+// This can be used to ensure a well-defined ordering of operations. Since each Server method is handled
+// asynchronously via a separate channel, if you call server.Publish and then immediately call server.Close,
+// there is no guarantee that the server execute the Close operation only after the event has been published.
+// If you instead call PublishWithAcknowledgement, and then read from the returned channel before calling
+// Close, you can be sure that the event was published before the server was closed.
+func (srv *Server) PublishWithAcknowledgment(channels []string, ev Event) <-chan struct{} {
+	ackCh := make(chan struct{}, 1)
 	srv.pub <- &outbound{
 		channels:       channels,
-		eventOrComment: comment{value: text},
+		eventOrComment: ev,
+		ackCh:          ackCh,
 	}
+	return ackCh
 }
 
-func replay(repo Repository, sub *subscription) {
-	for ev := range repo.Replay(sub.channel, sub.lastEventID) {
-		sub.out <- ev
+// PublishComment publishes a comment to one or more channels.
+func (srv *Server) PublishComment(channels []string, text string) {
+	srv.pub <- &outbound{
```

Comment:
Should this just construct the `eventOrComment` value, and just defer to `Publish`?

Reply:
There are two reasons not to do that. First, `Publish` specifically takes an `Event`; changing its signature would be a breaking change. Second, `eventOrComment` is deliberately not an exported interface, because it is really just `interface{}`— there aren't any actual methods that could enforce that it can only ever be an Event or a comment.